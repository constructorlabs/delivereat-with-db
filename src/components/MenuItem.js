import React from 'react'

import '../styles/MenuItem.scss'

class MenuItem extends React.Component{
  constructor(){
    super()

    this.handleAdd = this.handleAdd.bind(this)
    this.handleMaximise = this.handleMaximise.bind(this)
  }

handleAdd(event){
  event.preventDefault()
  this.props.addToOrder(this.props.menuItem, [], 1)
}

handleMaximise(event){
  event.preventDefault()
  if (this.props.menuItem.type === 'Burgers'){
    this.props.changeDisplay('maximised', this.props.menuItem)
  }
}

  render(){
    const backgroundStyle = {
      backgroundImage: `url(${this.props.menuItem.photo_url})`

    }

    return(
      <div onClick={this.handleMaximise} className="transparent-gradient">
      <li style={backgroundStyle} className="menu-item"><h5 className="menu-item__name">{this.props.menuItem.name}</h5>
         <span className="menu-item__price-add"> <span className="menu-item__price">Â£{this.props.menuItem.price}    </span>
         {this.props.menuItem.type !== 'Burgers'
          ? <a href="" onClick={this.handleAdd}><i className="fas fa-plus-circle plus-menu"></i></a>
          : null
        }
        </span>
      </li>
    </div>
    )
  }


}

export default MenuItem
