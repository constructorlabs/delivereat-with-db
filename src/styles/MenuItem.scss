.menu-item{
  font-family: 'Noto Sans', sans-serif;
  position: relative;
  max-width: 100%;
  height: 100%;
  list-style: none;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding: 1rem;
  margin-bottom: 0.5rem;
  background-size: cover;
  background-position: center;
  word-break: break-all;

  border: 1px solid black;
  border-radius: 2%;

  &:after{
    display: flex;
    z-index: 1;
    content: "";
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    position: absolute;
    background-image: linear-gradient(to top, rgba(255,0,0,0), #040404);
    opacity: .6;
    overflow: auto;
  }

  &__name{
    z-index: 100;
    margin-top: 0;
    color: white;
  }

  &__price-add{
    z-index: 100;
    display: flex;
    flex-direction: row;
    justify-content: space-between;


  }

  &__price{
    background-color: #73db3b;
    margin-right: 1rem;
    padding: 0.5rem;
  }
}


.plus-menu{
  color: #FFDA02;
  font-size: 1.5rem;
}
