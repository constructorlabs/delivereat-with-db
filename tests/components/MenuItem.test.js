import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MenuItem from '../../src/components/MenuItem';

const stage = 'basket';
const order = { id: 10, items: [[1, 1], [2, 3]] };
const itemToAdd = { id: 1, name: 'Espresso', description: 'Single shot, 30ml', price: '2.00' };
const itemToRemove = {
  id: 3,
  name: 'Macchiato',
  description: '60ml espresso + a dab of milk foam',
  price: '2.20',
};
const mockAddToOrder = jest.fn();
const mockRemoveFromOrder = jest.fn();

describe('Menu Item', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MenuItem
          stage={stage}
          order={order}
          addToOrder={mockAddToOrder}
          removeFromOrder={mockRemoveFromOrder}
          menuItem={itemToAdd}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('click adds item', () => {
    const wrapper = shallow(
      <MenuItem
        stage={stage}
        order={order}
        addToOrder={mockAddToOrder}
        removeFromOrder={mockRemoveFromOrder}
        menuItem={itemToAdd}
      />,
    );
    wrapper.find('.menu-item__add-button').simulate('click');
    expect(mockAddToOrder.mock.calls).toEqual([[1]]);
  });
  test('click removes item', () => {
    const wrapper = shallow(
      <MenuItem
        stage={stage}
        order={order}
        addToOrder={mockAddToOrder}
        removeFromOrder={mockRemoveFromOrder}
        menuItem={itemToRemove}
      />,
    );
    wrapper.find('.menu-item__remove-button').simulate('click');
    expect(mockRemoveFromOrder.mock.calls).toEqual([[3]]);
  });
});
