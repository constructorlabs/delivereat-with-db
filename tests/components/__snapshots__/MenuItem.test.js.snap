// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item matches the snapshot 1`] = `
<div
  className="menu-item menu-item--in-basket menu-item--greyed-out"
>
  <div
    className="menu-item__title"
  >
    <div
      className="menu-item__name"
    >
      Espresso
    </div>
    <div
      className="menu-item__buttons"
    >
      <button
        className="menu-item__remove-button menu-item__remove-button--in-basket"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-minus-circle"
        />
      </button>
      <div
        className="menu-item__count menu-item__count--in-basket"
      >
        1
      </div>
      <button
        className="menu-item__add-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-plus-circle"
        />
      </button>
    </div>
  </div>
  <div
    className="menu-item__description"
  >
    Single shot, 30ml
  </div>
  <div
    className="menu-item__price"
  >
    Â£2.00
  </div>
</div>
`;
